login/session.go

```diff
@@ -555,3 +584,28 @@ func (b *SessionBuilder) handleEventExpireOtherSessions(ctx *web.EventContext) (
 	web.AppendRunScripts(&r, web.Plaid().MergeQuery(true).Go())
 	return
 }
+
+func (b *SessionBuilder) handleEventExpireSession(ctx *web.EventContext) (r web.EventResponse, err error) {
+	msgr := i18n.MustGetModuleMessages(ctx.R, I18nAdminLoginKey, Messages_en_US).(*Messages)
+
+	user := login.GetCurrentUser(ctx.R)
+	if user == nil {
+		return r, errors.New("login: user not found")
+	}
+	isPublicUser := false
+	if b.isPublicUser != nil {
+		isPublicUser = b.isPublicUser(user)
+	}
+	if isPublicUser {
+		return r, perm.PermissionDenied
+	}
+	uid, tokenHash := ctx.R.FormValue("uid"), ctx.R.FormValue("token_hash")
+
+	if err = b.ExpireSession(uid, tokenHash); err != nil {
+		return r, err
+	}
+
+	presets.ShowMessage(&r, msgr.SuccessfullyExpiredSessions, "")
+	web.AppendRunScripts(&r, web.Plaid().URL("/"+uriNameLoginSessionsDialog).EventFunc(eventLoginSessionsDialog).MergeQuery(true).Go())
```

Comment:
@molon Could you please check if there’s a better implementation here? The purpose is to refresh the sessions dialog.